Increase simulation count as optimizer runs, up to a max set by user

xoth42 opened the pull request:
Set a max_simulation count, in the keyword arguments of 'multiple_simulations_with_history!' and the optimizer will increase the simulation count if needed up to that limit. Displays current and max.
<img width="659" height="457" alt="image" src="https://github.com/user-attachments/assets/09838833-db31-40fb-aece-67305677c192" />

For issue #11

xoth42 committed:
edge case fix for mutate

xoth42 committed:
Increase simulation count as optimizer runs, up to a max set by user

Krastanov reviewed (changes requested):
really neat logic!

Add a bit more clarity to the config docs and warning

- Specify that this is NOT "annealing", rather sim count goes up only when necessary
- Just overall make docs and warnings around this clearer

Krastanov commented:
feel free to merge after addressing comments

dependabot[bot] committed:
Bump actions/checkout from 4 to 5 (#37)

Bumps [actions/checkout](https://github.com/actions/checkout) from 4 to 5.
- [Release notes](https://github.com/actions/checkout/releases)
- [Changelog](https://github.com/actions/checkout/blob/main/CHANGELOG.md)
- [Commits](https://github.com/actions/checkout/compare/v4...v5)

---
updated-dependencies:
- dependency-name: actions/checkout
  dependency-version: '5'
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

xoth42 committed:
update to v0.1.2: Pluto reactivity, user control

Running the pluto notebook is a lot more enjoyable from the user side. Configuration changes are buffered by a PlutoUI.confirm query, meaning many variables can be changed before any simulation is run. Restarting/running the simulation also are now buffered with buttons.

Krastanov committed:
fix downgrade (#38)

* fix downgrade

* exclude Statistics from downgrade check

* bump downgrade bounds

xoth42 committed:
Merge branch 'master' into anealing-parameters

xoth42 committed:
syntax fix

xoth42 committed:
syntax fix for multiple steps
